manifests/charts/istio-control/istio-discovery/files/grpc-agent.yaml

```diff
@@ -197,7 +181,7 @@ spec:
     volumeMounts:
     - name: workload-socket
       mountPath: /var/run/secrets/workload-spiffe-uds
-    {{- if eq .Values.global.caName "GkeWorkloadCertificate" }}
+    {{- if or (isset .ObjectMeta.Annotations `security.cloud.google.com/use-workload-certificates`) (eq .Values.global.caName "GkeWorkloadCertificate") }}
```

Comment:
I don't know if we want it for the regular case. 

With gRPC, the intent is to eventually be 'agent-less' and 'injection-less' too - so users can just deploy a pod with a proxyless app and it would work. In this case they would use the annotation ( or whatever equivalent volumes are used for other cert providers ). 

Let's see how it works with grpc - and we can revisit if we should also add it to Istio. Since this is a documented feature 
in GKE users may set it anyways, so probably we need to deal with it - I suspect the naming of the volume to match
what they generate is handling the case.

Reply:
Removed it from the workload and volume - if the annotation is set, their injector will create the same thing.

For proxy - we still have an ordering issue, if their injector goes first istio-proxy will end up without the volume mount.

I'll keep playing with this on the grpc - right now we are not supporting the annotation, but will be useful to not break ( but not related to this PR )